lib/Dialect/TT/IR/TTOpsTypes.cpp

```diff
@@ -134,13 +147,72 @@ mlir::tt::SystemDescAttr::getFromPath(MLIRContext *context, std::string &path) {
       break;
     }
 
+    std::vector<tt::DataType> supported_data_types;
+    for (auto it : *(element->supported_data_types())) {
+      switch (it) {
+      case ::tt::target::DataType::Float32:
+        supported_data_types.push_back(tt::DataType::Float32);
+        break;
+      case ::tt::target::DataType::Float16:
+        supported_data_types.push_back(tt::DataType::Float16);
+        break;
+      case ::tt::target::DataType::BFloat16:
+        supported_data_types.push_back(tt::DataType::BFloat16);
+        break;
+      case ::tt::target::DataType::BFP_Float8:
+        supported_data_types.push_back(tt::DataType::BFP_Float8);
+        break;
+      case ::tt::target::DataType::BFP_BFloat8:
+        supported_data_types.push_back(tt::DataType::BFP_BFloat8);
+        break;
+      case ::tt::target::DataType::BFP_Float4:
+        supported_data_types.push_back(tt::DataType::BFP_Float4);
+        break;
+      case ::tt::target::DataType::BFP_BFloat4:
+        supported_data_types.push_back(tt::DataType::BFP_BFloat4);
+        break;
+      case ::tt::target::DataType::BFP_Float2:
+        supported_data_types.push_back(tt::DataType::BFP_Float2);
+        break;
+      case ::tt::target::DataType::BFP_BFloat2:
+        supported_data_types.push_back(tt::DataType::BFP_BFloat2);
+        break;
+      case ::tt::target::DataType::UInt32:
+        supported_data_types.push_back(tt::DataType::UInt32);
+        break;
+      case ::tt::target::DataType::UInt16:
+        supported_data_types.push_back(tt::DataType::UInt16);
+        break;
+      case ::tt::target::DataType::UInt8:
+        supported_data_types.push_back(tt::DataType::UInt8);
+        break;
+      }
+    }
+
+    std::vector<tt::DataTypeAttr> supported_data_types_attr;
+
+    for (auto it : *(element->supported_data_types())) {
+      supported_data_types_attr.push_back(
+          tt::DataTypeAttr::get(context, DataType(it)));
+    }
+
+    // populate a placeholder for supported tile sizes
+    SmallVector<tt::GridSizeAttr> supported_tile_sizes;
+    supported_tile_sizes.push_back(tt::GridSizeAttr::get(context, 4, 16));
+    supported_tile_sizes.push_back(tt::GridSizeAttr::get(context, 16, 16));
+    supported_tile_sizes.push_back(tt::GridSizeAttr::get(context, 32, 16));
+    supported_tile_sizes.push_back(tt::GridSizeAttr::get(context, 4, 32));
+    supported_tile_sizes.push_back(tt::GridSizeAttr::get(context, 16, 32));
+    supported_tile_sizes.push_back(tt::GridSizeAttr::get(context, 32, 32));
```

Comment:
We should parse these out of the flatbuffer.

Reply:
I agree, this doesn't look right. SystemDesc would be populated with values for supported tiile sizes, and this function here should be parsing/extracting them from the flatbuffer/systemdesc.